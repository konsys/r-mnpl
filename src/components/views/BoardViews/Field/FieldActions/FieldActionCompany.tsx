import { IFieldRent } from "../../../../../types/types";
import React from "react";

export const FieldActionCompany = (rent: IFieldRent) => {
  return (
    <div className="TableFieldcard-data-rows">
      <div>
        <div className="_title">
          <span>Базовая рента</span>
        </div>
        <div className="_value _type_money _base_rent">
          {rent && rent.baseRent}
        </div>
      </div>
      <div>
        <div className="_title">
          <span className="_star_small"></span>
        </div>
        <div className="_value _type_money _one_star_rent">
          {rent && rent.oneStar}
        </div>
      </div>
      <div>
        <div className="_title">
          <span className="_star_small"></span>
          <span className="_star_small"></span>
        </div>
        <div className="_value _type_money _two_star_rent">
          {rent && rent.twoStar}
        </div>
      </div>
      <div>
        <div className="_title">
          <span className="_star_small"></span>
          <span className="_star_small"></span>
          <span className="_star_small"></span>
        </div>
        <div className="_value _type_money _free_star_rent">
          {rent && rent.freeStar}
        </div>
      </div>
      <div>
        <div className="_title">
          <span className="_star_small"></span>
          <span className="_star_small"></span>
          <span className="_star_small"></span>
          <span className="_star_small"></span>
        </div>
        <div className="_value _type_money _four_star_rent">
          {rent && rent.fourStar}
        </div>
      </div>
      <div>
        <div className="_title">
          <span className="_star_big"></span>
        </div>
        <div className="_value _type_money _big_star_rent">
          {rent && rent.bigStar}
        </div>
      </div>
    </div>
  );
};
