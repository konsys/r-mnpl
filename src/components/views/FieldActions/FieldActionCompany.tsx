import React from "react";
import { IFieldRent } from "../../../types/BoardTypes";

export const FieldActionCompany = (rent: IFieldRent) => {
  return (
    <div className="TableFieldcard-data-rows">
      <div>
        <div className="_title">
          <span>Базовая рента</span>
        </div>
        <div className="_value _type_money">{rent && rent.baseRent}</div>
      </div>
      <div>
        <div className="_title">
          <span className="_star_small"></span>
        </div>
        <div className="_value _type_money">{rent && rent.oneStar}</div>
      </div>
      <div>
        <div className="_title">
          <span className="_star_small"></span>
          <span className="_star_small"></span>
        </div>
        <div className="_value _type_money">{rent && rent.twoStar}</div>
      </div>
      <div>
        <div className="_title">
          <span className="_star_small"></span>
          <span className="_star_small"></span>
          <span className="_star_small"></span>
        </div>
        <div className="_value _type_money">{rent && rent.freeStar}</div>
      </div>
      <div>
        <div className="_title">
          <span className="_star_small"></span>
          <span className="_star_small"></span>
          <span className="_star_small"></span>
          <span className="_star_small"></span>
        </div>
        <div className="_value _type_money">{rent && rent.fourStar}</div>
      </div>
      <div>
        <div className="_title">
          <span className="_star_big"></span>
        </div>
        <div className="_value _type_money">{rent && rent.bigStar}</div>
      </div>
    </div>
  );
};
