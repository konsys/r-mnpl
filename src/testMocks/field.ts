import { FieldStatus, FieldType, IField, IFieldAction } from "types/types";

import { FieldActions } from "components/views/BoardViews/Field/FieldActions/FieldAction";

// import {FieldStatus} from ""

export const testField: IField = {
  fieldId: 1,
  fieldPosition: 1,
  name: "testName",
  type: FieldType.COMPANY,
  price: {
    branchPrice: 1000,
    buyoutPrice: 900,
    pledgePrice: 500,
    startPrice: 2000,
  },
  rent: {
    baseRent: 200,
    monopolyRent: 250,
    oneStar: 300,
    twoStar: 1000,
    freeStar: 1500,
    fourStar: 2000,
    bigStar: 3000,
  },
};

export const testOwnedField: IField = {
  status: {
    level: 0,
    branches: 0,
    isMonopoly: false,
    fieldId: 1,
    userId: 1,
    mortgaged: 0,
    fieldActions: [IFieldAction.MORTGAGE],
  },
  fieldId: 1,
  fieldPosition: 1,
  name: "testName",
  type: FieldType.COMPANY,
  price: {
    branchPrice: 1000,
    buyoutPrice: 900,
    pledgePrice: 500,
    startPrice: 2000,
  },
  rent: {
    baseRent: 200,
    monopolyRent: 250,
    oneStar: 300,
    twoStar: 1000,
    freeStar: 1500,
    fourStar: 2000,
    bigStar: 3000,
  },
};

export const testJailField: IField = {
  fieldId: 10,
  fieldPosition: 10,
  name: "jail",
  type: FieldType.JAIL,
  fieldCorner: 2,
  fieldLine: 1,
  isJail: true,
};
