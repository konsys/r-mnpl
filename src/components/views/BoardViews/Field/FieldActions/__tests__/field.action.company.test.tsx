import { FieldActionCompany } from "../FieldActionCompany";
import React from "react";
import { shallow } from "enzyme";
import { testRent } from "testMocks/field.rent";

describe("Field action auto test", () => {
  it("should render properly", () => {
    expect(shallow(<FieldActionCompany {...testRent} />)).toMatchSnapshot();
  });
  it("should show branches price", () => {
    const rendered = shallow(<FieldActionCompany {...testRent} />);
    expect(rendered.find("._base_rent").text()).toBe(
      testRent.baseRent.toString()
    );
    expect(rendered.find("._one_star_rent").text()).toBe(
      testRent.oneStar.toString()
    );
    expect(rendered.find("._two_star_rent").text()).toBe(
      testRent.twoStar.toString()
    );
    expect(rendered.find("._free_star_rent").text()).toBe(
      testRent.freeStar.toString()
    );
    expect(rendered.find("._four_star_rent").text()).toBe(
      testRent.fourStar.toString()
    );
    expect(rendered.find("._big_star_rent").text()).toBe(
      testRent.bigStar.toString()
    );
    expect(rendered.find("._star_small")).toHaveLength(10);
    expect(rendered.find("._star_big")).toHaveLength(1);
  });
});
