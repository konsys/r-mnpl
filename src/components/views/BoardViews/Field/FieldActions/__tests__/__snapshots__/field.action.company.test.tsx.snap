// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field action auto test should render properly 1`] = `ShallowWrapper {}`;
